### PR TITLE
Support metrics

### PR DESCRIPTION
**Pull request by mdesmet**
### Problem

Support metrics in lineage panel and tree view

### Solution

Only semantic_model is supported

### Screenshot/Demo

image

Currently incorrectly shown as view @AdiGajbhiye 

### How to test

Add metrics to your project for example:

```
semantic_models:
  #The name of the semantic model.
  - name: orders
    defaults:
      agg_time_dimension: ordered_at
    description: |
      Order fact table. This table is at the order grain with one row per order. 
    #The name of the dbt model and schema
    model: ref('orders')
```

## Checklist

- [ ] I have run this code and it appears to resolve the stated issue
- [ ] `README.md` updated and added information about my change

### CODE REVIEW

**Review by mdesmet — COMMENTED** [review]
A metric is not a view
The icon should be updated.
Details should not be displayed for metrics.

File: src/manifest/parsers/graphParser.ts [hunk 1]
```diff
@@ -68,7 +85,33 @@ export class GraphParser {
...
+        const currentNodes = unique(nodes)
+          .map(
+            this.mapToNode(
+              sourceMetaMap,
+              nodeMetaMap,
+              testMetaMap,
+              metricMetaMap,
+            ),
+          )
           .filter((n) => n instanceof Test)
```
Comment (AdiGajbhiye) [inline src/manifest/parsers/graphParser.ts:115]: Shouldn't this be `Metric`

File: src/webview_provider/newLineagePanel.ts [hunk 2]
```diff
@@ -710,11 +706,24 @@ export class NewLineagePanel implements LineagePanelView {
...
+    if (nodeType === DBTProject.RESOURCE_TYPE_METRIC) {
+      return {
+        table: key,
+        label: splits[2],
+        url: tableUrl,
+        upstreamCount,
+        downstreamCount,
+        nodeType,
+        materialization: undefined,
+        tests: (graphMetaMap["tests"].get(key)?.nodes || []).map((n) => {
```
Comment (saravmajestic) [inline src/webview_provider/newLineagePanel.ts:718]: semantic model could have test?

**Review by saravmajestic — COMMENTED** [review]
LGTM, except Aditya's comment

File: src/webview_provider/newLineagePanel.ts (hunk 2 above)
Comment (mdesmet) [reply src/webview_provider/newLineagePanel.ts:718]: They don't, also exposures don't have tests. Will remove.

**Pull request merged**